odoo/addons/base/models/ir_attachment.py

```diff
@@ -630,11 +635,11 @@ def _post_add_create(self):
     def generate_access_token(self):
         tokens = []
         for attachment in self:
-            if attachment.access_token:
-                tokens.append(attachment.access_token)
+            if attachment.sudo().access_token:
```

Comment:
As this is a public method, it will be necessary to check that the user has the right to generate the access token for this document. Otherwise, an average user can use this method to create access for a private document.